docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+int s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac hmac);
+int s2n_connection_append_psk(struct s2n_connection *conn, struct s2n_psk *psk);
+int s2n_config_set_psk_mode(struct s2n_config *config, s2n_psk_mode mode);
+int s2n_connection_set_psk_mode(struct s2n_connection *conn, s2n_psk_mode mode);
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length);
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+struct s2n_offered_psk* s2n_offered_psk_new();
+int s2n_offered_psk_free(struct s2n_offered_psk **psk);
+int s2n_offered_psk_get_identity(struct s2n_offered_psk *psk, uint8_t** identity, uint16_t *size);
+bool s2n_offered_psk_list_has_next(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_next(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+int s2n_offered_psk_list_reread(struct s2n_offered_psk_list *psk_list);
+int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struct s2n_offered_psk *psk);
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, void *context,
+                                          struct s2n_offered_psk_list *psk_list);
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb, void *context);
+```
+
+### Configuring External Pre-Shared Keys
+
+Use the following APIs to configure and use external pre-shared keys in the > =TLS1.3 connections. An external PSK is an out-of-band provisioned pre-shared key (PSK) established externally using a secure mutually agreed upon mechanism.
+
+To create an external PSK a caller must call the `s2n_external_psk_new` API.**s2n_external_psk_new** creates a new s2n external pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default pre-shared key hash algorithm. Note that the obfuscated ticket age will always be zero for external PSKs. Use `s2n_psk_free` to free the memory allocated to the s2n external pre-shared key object created by this API.
+
+**s2n_psk_free** frees the memory associated with the `s2n_psk` object.
+
+After creation of the external PSK object use `s2n_psk_set_identity`, `s2n_psk_set_secret` to set the PSK Identity and PSK secret. To override the default PSK Hash Algorithm  use `s2n_psk_set_hmac`.
+
+**s2n_psk_set_identity** sets the pre-shared key (PSK) identity for a given s2n_psk object. The pre-shared key identity is a unique identifier for the pre-shared secret. The Pre-shared key identity is transmitted over the network unencrypted and is a non-secret value. Be sure to not include any confidential information as a part of the pre-shared key identity. Note that the identity value is copied into s2n-tls memory and the user is responsbile for freeing the memory  associated with the `identity` input. Be sure to call `free(identity)` after this API call. 
```

Comment:
We either need to say pre-shared key every time, or accept that the intro explained "PSK" and we can now use PSK. Mixing them and repeating the (PSK) seems messy.

```suggestion
**s2n_psk_set_identity** sets the identity for a given PSK. The identity must be unique for all PSKs supported by a server or client. The identity is transmitted over the network unencrypted and is a non-secret value, so be sure not to incorporate any confidential information. Note that the identity value is copied into s2n-tls memory and the user is responsible for freeing the memory associated with the input.
```